src/credentials-manager/index.js

```diff
@@ -0,0 +1,147 @@
+   * @param {String} credentials.refreshToken optional - used to refresh access token
+   * @param {String} credentials.scope optional - represents the scope of the current token
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async saveCredentials(credentials = {}) {
+    const validateKeys = ['idToken', 'accessToken', 'tokenType', 'expiresIn'];
+    validateKeys.forEach(key => {
+      if (!credentials[key]) {
+        const json = {
+          error: 'a0.credential_manager.invalid_input',
+          error_description: `${key} cannot be empty`,
+          invalid_parameter: key,
+        };
+        throw new CredentialsManagerError({json, status: 0});
+      }
+    });
+    try {
+      await this._ensureCredentialManagerIsInitialized();
+      return await this.Auth0Module.saveCredentials(credentials);
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Gets the credentials that has already been saved
+   *
+   * @param {String} scope optional - the scope to request for the access token. If null is passed, the previous scope will be kept.
+   * @param {String} minTtl optional - the minimum time in seconds that the access token should last before expiration.
+   * @param {Object} parameters optional - additional parameters to send in the request to refresh expired credentials.
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async getCredentials(scope, minTtl = 0, parameters = {}) {
+    try {
+      await this._ensureCredentialManagerIsInitialized();
+      const credentials = await this.Auth0Module.getCredentials(
+        scope,
+        minTtl,
+        parameters,
+      );
+      return credentials;
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Enables Local Authentication (PIN, Biometric, Swipe etc) to get the credentials
+   *
+   * @param {String} title optional - the text to use as title in the authentication screen. Passing null will result in using the OS's default value in Android and "Please authenticate to continue" in iOS.
+   * @param {String} description Android Only - optional - the text to use as description in the authentication screen. On some Android versions it might not be shown. Passing null will result in using the OS's default value.
+   * @param {String} cancelTitle iOS Only - optional - the cancel message to display on the local authentication prompt.
+   * @param {String} fallbackTitle iOS Only - optional - the fallback message to display on the local authentication prompt after a failed match.
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async requireLocalAuthentication(
```

Comment:
How do we plan on surfacing this to the end developer? Should this be another method on the hook, or a prop on `Auth0Provider`?

In Flutter we have it as an argument on the `Auth0` class, which I suppose is the equivalent of `Auth0Provider` in our case, but also we have a [load of different options](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter_platform_interface/lib/src/credentials-manager/options/local_authentication_options.dart) so may not actually be suitable for props.

Or should the end developer have access to the credentials manager instance so they can just call `requireLocalAuthentication` directly?

My preference would be to expose it as a method on the hook and just pass it through.

Reply:
I would agree with it. The best is to have it as a method for both iOS and Android. This is not just because of the different options we provide, but also currently both iOS and Android don't have an option to disable this once enabled.